sdk/src/main/java/com/hcaptcha/sdk/HCaptchaDialogFragment.java

```diff
@@ -156,6 +157,20 @@ private void setupWebView(final WebView webView) {
         webView.loadUrl("file:///android_asset/hcaptcha-form.html");
     }
 
+    private OnFailureListener getFailureListener(boolean resetOnError) {
+        if (resetOnError) {
+            return new OnFailureListener() {
+                @Override
+                public void onFailure(@NonNull final HCaptchaException exception) {
+                    if (exception.getHCaptchaError() == HCaptchaError.SESSION_TIMEOUT) {
```

Comment:
It looks like redundant handler as there is already one on line 207. This if statement can be moved there.

Reply:
I tested with updated existing handler and seems to work just fine.

```java
// line 191
@Override
public void onFailure(@NonNull final HCaptchaException hCaptchaException) {
    final boolean silentRetry = this.resetOnTimeout && hCaptchaException.getHCaptchaError() == HCaptchaError.SESSION_TIMEOUT;
    if (isAdded() && !silentRetry) {
        dismiss();
    }
    handler.post(new Runnable() {
        @Override
        public void run() {
            if (silentRetry) {
                // Re-open the challenge
                webView.loadUrl("javascript:hcaptcha.execute()");
            } else {
                hCaptchaDialogListener.onFailure(hCaptchaException);
            }
        }
    });
}
```

Reply:
Updated